torch/_inductor/lowering.py

```diff
@@ -2172,9 +2172,13 @@ def fn(idx):
         assert len(idx) == len(output_size)
         assert len(indices_loaders) == len(indexed_size)
         new_index = [
-            ops.indirect_indexing(
-                loader(idx[start_offset:end_offset]), size, check=check
+            (
+                ops.indirect_indexing(
+                    loader(idx[start_offset:end_offset]), size, check=check
+                )
+                + size
             )
+            % size
```

Comment:
This doesn't really work because `ops.indirect_indexing` will assert if the value is less than 0 at runtime. You also can't just move the modulo before the `ops.indirect_indexing` call because you'd always pass the assert, even for invalid indices! I think you really need something like 
```python
size_ir = ops.index_expr(size, torch.int64)
zero = ops.constant(0, dtype)
ops.indirect_indexing(ops.where(x < zero, x + size_ir, x), size)
```

Reply:
We didn't change things before `ops.indirect_indexing`, but correct the index after `ops.indirect_indexing`.